File: move_zeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int j = 0;
        for(int i = 0; i < nums.size()-1; i++){
            if(nums[i] == 0){
                if(j < i)
                    j = i + 1;
                while(j < nums.size() && nums[j] == 0) j++;
                if(j != nums.size() && j > i)
                    swap(nums[i], nums[j]);
            } else j = max(j,i);
        }
        return ;
    }
};

File: move_zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        pt0 = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[pt0], nums[i] = nums[i], nums[pt0]
                pt0 += 1